From 79b101a94f1fa8a3753535bbff961992c65885e5 Mon Sep 17 00:00:00 2001
From: ziv zaslavski <zzaslavski@habana.ai>
Date: Mon, 09 Jan 2023 10:59:43 +0200
Subject: [PATCH] [SV-769] ppboot: gaudi2: pci: fix race of fw_fsm_size write

Add delay between writing FSM_SIZE and FW_FSM_EN to fix race between
propagation of fsm_size and latching it in the fsm.

Change-Id: I70f12129528165e5c7e36b8b38506704683e6c50
Signed-off-by: ziv zaslavski <zzaslavski@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 2ce57cc..c4eb6a6 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -291,6 +291,12 @@
 	regmap_writel(FW_WORD_NUM, mmPCIE_WRAP_PHY_FW_FSM_SIZE);
 
 	/*
+	 * Delay for the value to seep through the clock domains and get to the FSM
+	 * As the PHY_FW_FSM_SIZE and the PHY_FW_FSM pass through different clock domains
+	 */
+	udelay(1);
+
+	/*
 	 * PHY FW FSM EN and Direct FW write mechanism enable
 	 * FW FSM assumes that the data is present in the SRAM
 	 * The FSM will wait for SRAM init done before updating
